### PR TITLE
feat: add i18n support to notifications

### PR DESCRIPTION
**Pull request by dolcalmi**
- Add locales support to notifications
- Add default locale and display currency config

### CODE REVIEW

**Review by UncleSamtoshi — CHANGES_REQUESTED** [review]

**Comment by UncleSamtoshi** [comment]
DisplayCurrenyConverter currently assumes that the [display currency is USD](https://github.com/GaloyMoney/galoy/blob/feat-notifications-currency/src/domain/fiat/display-currency.ts) in ```fromCents```. I think we need to remove this method or throw an error if the display currency is not Usd.

**Comment by UncleSamtoshi** [comment]
A couple questions:
- Can the display currency ever change? If so, what happens to any transaction history that has a unitless display currency?
- How do we ensure that the Price server is providing the price for the display currency that is configured in the backend? Do they use the same configuration?

**Comment by UncleSamtoshi** [comment]
> DisplayCurrenyConverter currently assumes that the [display currency is USD](https://github.com/GaloyMoney/galoy/blob/feat-notifications-currency/src/domain/fiat/display-currency.ts) in `fromCents`. I think we need to remove this method or throw an error if the display currency is not Usd.

You could change ```fromCents(amount)``` to be ```fromSatsAndCents(amountSats, amountCents) { if (displayCurrency === "USD") { return amountCents} else { return amountSats*displayCurrencyPerSat }}```

**Comment by dolcalmi** [comment]
> * Can the display currency ever change? If so, what happens to any transaction history that has a unitless display currency?

we will need to add display currency to db but this will be done after integrate price history PR https://github.com/GaloyMoney/galoy/pull/1162
> * How do we ensure that the Price server is providing the price for the display currency that is configured in the backend? Do they use the same configuration?

if price server does not support the configured display currency then we will get a critical error and probably server will not start
> You could change `fromCents(amount)` to be `fromSatsAndCents(amountSats, amountCents) { if (displayCurrency === "USD") { return amountCents} else { return amountSats*displayCurrencyPerSat }}`

I guess this is not enough, we will need to change `UsdCents` type to something like `Cents` and probably rename other types too but this will be done in the PR that will add support to multiple currencies

File: src/config/locales/en.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
+{
+  "notification": {
+    "balance": {
+      "fiat": "Your balance is %s%s (%s sats)",
```
[the remark this answers is not in the document]
Comment (dolcalmi) [reply src/config/locales/en.json]: Fixed https://github.com/GaloyMoney/galoy/pull/1178/commits/be618ce88da2ebd52032440bb9cb37f7f8d22ef8

**Review by UncleSamtoshi — APPROVED** [review]

**Pull request merged**